### ISSUE TITLE
Antimarscirce: castlings are not executed

### ISSUE THREAD

**Issue by thomas-maeder** [thomas-maeder/popeye#490]
From yesterday's solving tourney in Andernach:

BeginProblem
Stipulation h#2
Condition AntiMarsCirce
Pieces
  black  Kf6 Bg1 Rh5 Pe6d7
  white  Kf1 Rh1 Sh4 Bf5
EndProblem

Popeye only finds
  1.Rh5*h4 Bf5*e6   2.Rh4-h8-e8 Rh1-h1-h6 #
but neither the second author's solution
1.Rh5*f5+ Kf1*g1 2.Rf5-a8-e8 0-0 #
nor my cook
1.0-0-0 Rh1-h2 2.R-h8-e8 R-a1-a8 #

**Comment by thomas-maeder** [comment]
This may be a duplicate of bug #489

**Comment by WalterL-wL** [comment]
Indeed this is a duplicate of my #489.
How would you understand the wording of your indicated 3rd solution "...nor my cook: 1.0-0-0..."? Do you think the black castling is legal when BOTH pieces have to be reborn in advance in the SAME non-capturing move, although castling by definition is a king's move only? So, what would be the justification for the rook's advance rebirth? - Thanks!

**Comment by WalterL-wL** [comment]
Considering your III. solution with 1.0-0-0:
1.Kf6[→e8],Rh5[→a8]-0-0-0 **Rh1[→h1]-h2**  2.Rd8[→h8]-e8 Rh2[→a1]-a8#

this seems to result in four *subsidiary* cooks:
a) 'coz 1... **Rh1×Bg1**  2.Rd8[→h8]-e8 Rg1[→a1]-a8#
b+c) 'coz 1... **Kf1[→e1]-d1,d2**  2.Rd8[→h8]-e8 **Bf5[→f1]-a6#**
d) 'coz 1... **Kf1×Bg1**  2.Rd8[→h8]-e8 **Bf5[→f1]-a6#**

**Comment by WalterL-wL** [comment]
Though I personally feel in Antimarscirce this kind of extensive interpretation of the scope of the king's move in castling (including advance-rebirth also of the rook) is pushed too far, it still is an interesting and provoking thought! Referring to my former comments (especially concerning those subsidiary cooks in III, also in equally adapting my w0-0 solution.) I'd like to suggest a *little modification of the diagram*, this should result in 3 "clean" solutions with no cooks under this extensive interpretation: **shift Rh1→b5!**
3 Solutions:
I) 1.Rh5×h4...  2... Rb5[→h1]-h6#
II) 1.Rh5×f5...  2... Kg1[→e1],Rb5[→h1]-0-0# (!!)
III) 1.Kf6[→e8],Rh5[→a8]-0-0-0 Rb5[→h1]-h2  2... Rh2[→a1]-a8#

**Comment by WalterL-wL** [comment]
Sorry, further amendments:
With a diagram **shift Rh1→b5** you'll even have a nice 4th solution:
IV) 1.e6×Lf5 K×Lg1  2.Rh5[→a8]-e8 R×f5#

If you only want those former 3 solutions, then you'd better **shift Rh1→d3**

If **shift Rh1→c4** you will only have 2 solutions (no b0-0-0 allowed)

If **shift Rh1→a6(or)e2** there's another way to imply b0-0-0 in a 4th solution:
IV) 1.Kf6[→e8],Rh5[→a8]-0-0-0 Ra6(or)e2×e6  2.Rd8[→h8]-e8 Re6×Re8#

**Comment by WalterL-wL** [comment]
*A big, big sorry!* The suggested versions **wRh1→b5** as well as **wRh1→d3** have cooks! Only the version **wRh1→a6(or)e2** seems to show the correct 4 solutions...

**Comment by thomas-maeder** [comment]
I called it "my cook" because I was the only participant who found it at the Andernach solving tourney.

The chess rules define castling as a king move in the context of the touch-move rule; i.e. if a player does an illegal castling, he is required to move his king instead, not the rook.
This rule isn't relevant in the context of chess composition; e.g. in Madrasi, castling isn't possible with a paralysed rook.

**Comment by thomas-maeder** [comment]
> this seems to result in four subsidiary cooks:

Indeed. I should have found R*g1 as alternative to Rh2; if I had given this as an additional solution, I might have won the Andernach solving tourney :-)

**Comment by WalterL-wL** [comment]
Thanks! As I understand you see your extended interpretation with this 3rd solution b0-0-0 as correct, so the original two solutions show a cook - I'm still undecided but rather open minded on this. I would fully agree if this will finally be a wider accepted interpretation, currently there still seems to be some opposite views on that, even some written reactions to Andernach 2024 seem to be announced... One could even refer to FIDE rule 3.8.2: "**Castling** - This is a move of the king **and** either rook of the same colour along the player’s first rank, counting as a **single move** of the king..."... but there's also this (orthodox) restriction "*along the player's first rank*"...? Of course FIDE doesn't set the legal context for chess composition, but where's THE genuine link then?

Maybe there's another unresolved difference: In contrast to your comment, according to Märchenschachlexikon, in **Madrasi** castling with a paralysed rook IS explicitely allowed! In **Chamäleonschach** castling doesn't change the rook to queen, *because* it is seen as a king's move. Whereas in **Einsteinschach** castling DOES change the rook to bishop (or queen in Reverse Einsteinschach). Especially this latter rule might create some rather quirky moves of the rook as a proper participant in castling with such an extended interpretation, because a rook(a3) shouldn't be able to move non-capturing "as a rook" in his advance-rebirth to a1, because he would be changed to bishop in this advance-move - what kind of move should it be else!? So, would it require a queen(a3 or even a4???) to move to a1 (becoming a rook in this advance-move) for executing w0-0-0? And as you such define the rook's move as a separate subpart of the move, then after castling again the rook would automatically change to bishop in Einsteinschach... Lots of unclear consequences with such an extended interpretation in my opinion...

**Comment by WalterL-wL** [comment]
On second and third thoughts I want to *withdraw* my apprehension of **quirky moves** of a rook in a combination (Reverse) Einsteinschach + Antimarscirce in this extended interpretation of castling! The superordinate rule of Antimarscirce defines that rebirth already happens in advance of the underlying non-capturing move, and on reference to Marscirce the rebirth square is defined by the current square before rebirthing... So, only a w rook proper (dark square) can be advance-reborn on a1 for performing w0-0-0 (the rook would finally be changed to bishop / queen only after castling). No problem!

**Comment by thomas-maeder** [comment]
@WalterL-wL Popeye now finds
```
  1.Rh5*h4 Bf5*e6   2.Rh4-h8-e8 Rh1-h1-h6 #
  1.Rh5*f5 + Kf1*g1   2.Rf5-a8-e8 0-0 #
  1.0-0-0 Kf1*g1   2.Rd8-h8-e8 Bf5-f1-a6 #
  1.0-0-0 Kf1-e1-d2   2.Rd8-h8-e8 Bf5-f1-a6 #
  1.0-0-0 Kf1-e1-d1   2.Rd8-h8-e8 Bf5-f1-a6 #
  1.0-0-0 Rh1*g1   2.Rd8-h8-e8 Rg1-a1-a8 #
  1.0-0-0 Rh1*g1   2.Rd8-h8-e8 Rg1-a1-c1 #
  1.0-0-0 Rh1-h1-h2   2.Rd8-h8-e8 Rh2-a1-a8 #
```

**Comment by WalterL-wL** [comment]
Hi Thomas,

great (and super-quick), ok! 

But I cannot fully understand the scope of your NB (... *to disambiguate rook rebirths for castlings*). Yes, insofar as castling per se is concerned it is perfect now: 2... [wKg1→e1][wRc4→h1]0-0 (before it was only 0-0). But at the same time (compare with yesterday's presentation style of 22. Apr., 19:06) you ***removed*** the advance rebirth square from all the other moves in the notations (e.g. now it is 2.Rh4-e8, yesterday it still was 2.Rh4-**h8**-e8). Is this intentional, and would it affect ALL advance rebirth notations everywhere? Actually I'd prefer the inclusion of the advance rebirth square (see also my comments to the inclusion of make&take interim squares).

THANKS, Walter

**Comment by thomas-maeder** [comment]
Since you insist...

**Comment by WalterL-wL** [comment]
> Since you insist...

Hi Thomas,

Thanks! I do hope my "insisting" should not characterize my (insignificant) personal view on this issue only, but provide a time-tested standard for a simpler traceability with these sometimes "confusing" advance rebirth moves...

Best, Walter

**Comment by thomas-maeder** [comment]
In fact, I had decided to leave out the intermediate squares without deep thought. Experience tells that sooner or later, decisions like this don't go unpunished.

I should have added a smiley to my comment.

**Comment by WalterL-wL** [comment]
no problem, even without a smiley...;)..., thanks!
